packages/react-navigation/lib/navigation-context.tsx

```diff
@@ -63,7 +65,14 @@ export class NavigationContextProvider extends React.Component<Props, State> {
   componentDidUpdate(prevProps: Props) {
     const { currentRoute, client } = this.props;
 
-    if (currentRoute && currentRoute !== this.state.previousRoute) {
+    if (currentRoute && currentRoute !== prevProps.currentRoute) {
+
+      // If there is already an active navigation span, end it with an 
+      // invalid time to cause it to be discarded from the context stack.
+      if (this.currentSpan) {
+        this.currentSpan.end(DISCARDED);
+      }
```

Comment:
I don't think ending the span with an end time of -1 will actually 'discard' the span? i.e. it will get popped off the context stack because it's been ended, but I think it will still get processed and sent...